vllm/worker/hpu_model_runner.py

```diff
@@ -354,6 +363,42 @@ def __init__(self, model, vllm_config, layer_names, is_causal, sampler):
             self.model.visual = htorch.hpu.wrap_in_hpu_graph(
                 self.model.visual, disable_tensor_cache=True)
 
+        # TODO : right now just enabling it keeping gemma3 in mind
+        if htorch.utils.internal.is_lazy() and is_gemma3(self.model):
+            logger.info("[Multimodal] Wrapping Visual Model")
+            self.model.vision_tower = htorch.hpu.wrap_in_hpu_graph(
+                self.model.vision_tower, disable_tensor_cache=True)
+            self.model.multi_modal_projector = htorch.hpu.wrap_in_hpu_graph(
+                self.model.multi_modal_projector, disable_tensor_cache=True)
```

Comment:
is this part related to sliding window? What happened if we don't wrap hpu separately?

Reply:
this is not part of sliding window, its for vision tower bucket handle